app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -197,11 +204,17 @@ def process_availability(availability)
 
       def parse_availability(availabilities, resources_by_metric_id)
         processed_availabilities_ids = availabilities.map do |availability|
-          resources_by_metric_id[availability['id']][:status] = process_availability(availability['data'].first)
+          availability_status = process_availability(availability['data'].first)
+          resources_by_metric_id[availability['id']].each do |resource|
+            resource[:status] = availability_status
+          end
           availability['id']
         end
         (resources_by_metric_id.keys - processed_availabilities_ids).each do |metric_id|
-          resources_by_metric_id[metric_id][:status] = process_availability nil
+          availability_status = process_availability nil
```

Comment:
Minor one, what about to not pass nil as argument and take is as default in the process_availability function.
It is more a style comment, no blocker.